VimeoUpload/Upload/Networking/New Upload (Private)/VimeoSessionManager+Upload.swift

```diff
@@ -38,70 +38,58 @@ public typealias UploadParameters = [String: Any]
 
-        let task = self.dataTask(with: request as URLRequest, completionHandler: { [weak self] (response, responseObject, error) -> Void in
-            
+        return self.request(request) { [jsonResponseSerializer] (sessionManagingResult: SessionManagingResult<JSON>) in
             // Do model parsing on a background thread
-            DispatchQueue.global(qos: .default).async(execute: { [weak self] () -> Void in
-                guard let strongSelf = self else
-                {
-                    return
-                }
-                
-                do
-                {
-                    let video = try (strongSelf.responseSerializer as! VimeoResponseSerializer).process(videoResponse: response, responseObject: responseObject as AnyObject?, error: error as NSError?)
-                    completionHandler(video, nil)
-                }
-                catch let error as NSError
-                {
-                    completionHandler(nil, error)
+            DispatchQueue.global(qos: .default).async(execute: {
+                switch sessionManagingResult.result {
+                case .failure(let error):
+                    completionHandler(nil, error as NSError)
+                case .success(let json):
+                    do {                        
+                        let video = try jsonResponseSerializer.process(
+                            videoResponse: sessionManagingResult.response,
+                            responseObject: json as AnyObject,
+                            error: nil
+                        )
+                        completionHandler(video, nil)
+                    } catch {
+                        completionHandler(nil, error as NSError)
+                    }
                 }
             })
-        })
-        
-        task.taskDescription = UploadTaskDescription.CreateVideo.rawValue
-        
-        return task
+        }
+
     }
     
-    func createVideoDownloadTask(url: URL, videoSettings: VideoSettings?, uploadParameters: UploadParameters) throws -> URLSessionDownloadTask
+    func createVideoDownloadTask(url: URL, videoSettings: VideoSettings?, uploadParameters: UploadParameters) throws -> Task?
     {
-        let request = try (self.requestSerializer as! VimeoRequestSerializer).createVideoRequest(with: url, videoSettings: videoSettings, uploadParameters: uploadParameters)
-        
-        let task = self.downloadTask(with: request as URLRequest, progress: nil, destination: nil, completionHandler: nil)
-        
-        task.taskDescription = UploadTaskDescription.CreateVideo.rawValue
-        
+        let request = try self.jsonRequestSerializer.createVideoRequest(with: url, videoSettings: videoSettings, uploadParameters: uploadParameters) as URLRequest        
+        let task = self.download(request) { _ in }
```

Comment:
Wonder if completion should be optional

Reply:
Since this is on the `VimeoNetworking` side I'd prefer to not make this optional just because of this use cause on Upload.